intercept/src/dispatch.cpp

```diff
@@ -2740,20 +2752,27 @@ CL_API_ENTRY cl_int CL_API_CALL CLIRN(clSetKernelArg)(
         GET_ENQUEUE_COUNTER();
 
         std::string argsString;
-        if( pIntercept->config().CallLogging )
+        if( pIntercept->config().CallLogging ||
+            (pIntercept->config().DumpReplayKernelEnqueue != -1) ||
+            (pIntercept->config().DumpReplayKernelName != "") )
         {
-            pIntercept->getKernelArgString(
-                arg_index,
-                arg_size,
-                arg_value,
-                argsString );
+        pIntercept->getKernelArgString(
+            arg_index,
+            arg_size,
+            arg_value,
+            argsString );
         }
         CALL_LOGGING_ENTER_KERNEL(
             kernel,
             "kernel = %p, %s",
             kernel,
             argsString.c_str() );
 
+        if( argsString.find( "CL_SAMPLER_NORMALIZED_COORDS" ) != std::string::npos && arg_value != nullptr )
+        {
+            // This argument is a sampler, dump it
+            pIntercept->saveSampler( kernel, arg_index, argsString );
+        }
```

Comment:
Performance impact of this should be zero for all intents and purposes, but fair, not the most elegant solution.

I have seen the sampler map, but it doesn't save what the arg_index is! If you want I can change it, so that it also saves it and then convert to a string when it has to be dumped. That does sound better, I agree.

Reply:
Aah, now I remember again: When we create a sampler and put it in the sampler map, we have no way of knowing which argument index it will have until we set it during `clSetKernelArg(...)`. So if there are multiple samplers in the sampler map, we have no way of knowing which one to dump and at which argument index.

I don't see a way around this problem, without making the code much more complex... It's a shame that the spec uses a dedicated type for a sampler, instead of just a bitset with a "classic" type such as in a kernel, then no special code would've been necessary :( 

I've added a check that we will only do the "is a sampler" test when we have the dumping for replay functionality enabled.